arbnode/dataposter/data_poster.go

```diff
@@ -285,7 +329,11 @@ func externalSigner(ctx context.Context, opts *ExternalSignerCfg) (signerFn, com
 			return nil, fmt.Errorf("unmarshaling signed transaction: %w", err)
 		}
 		hasher := types.LatestSignerForChainID(tx.ChainId())
-		if h := hasher.Hash(args.ToTransaction()); h != hasher.Hash(signedTx) {
+		gotTx, err := args.ToTransaction()
+		if err != nil {
+			return nil, fmt.Errorf("converting transaction arguments into transaction: %w", err)
+		}
+		if h := hasher.Hash(gotTx); h != hasher.Hash(signedTx) {
```

Comment:
I think this should compare hash(tx) to hash(signedTx). I don't think gotTx is relevant here

Reply:
In practice this shouldn't matter though